src/invoices/InvoiceForm/InvoiceLinksForm/InvoiceLinksForm.js

```diff
@@ -25,23 +31,54 @@ const InvoiceLinksForm = () => {
             data-test-invoice-form-link-name
             xs={4}
           >
-            <Field
-              component={TextField}
-              label={linkNameLabel}
-              name={`${elem}.name`}
-              disabled={isDisabled}
-            />
+            {isDisabled
+              ? (
+                <KeyValue
+                  label={linkNameLabel}
+                  value={fields.value[index].name}
```

Comment:
IMHO, this makes sense to write above, near with `const isDisabled = Boolean(fields.value[index].id);`
something like `cons {id, name, ...} = fields.value[index];